src/methods/wavelet_based.jl

```diff
@@ -1,67 +1,128 @@
 
-A wavelet surrogate generated by taking the maximal overlap discrete 
-wavelet transform (MODWT) of the signal, shuffling detail 
-coefficients at each dyadic scale using the provided `surromethod`,
-then taking the inverse transform to obtain a surrogate.
-
-## Coefficient shuffling method
-
-In contrast to the original 
-implementation where IAAFT is used, you may choose to use any surrogate 
-method from this package to perform the randomization of the detail 
-coefficients at each dyadic scale. Note: The iterative procedure after 
-the rank ordering step (step [v] in [^Keylock2006]) is not performed in 
-this implementation.
-
-If `surromethod == IAAFT()`, the wavelet surrogates preserves the local 
-mean and variance structure of the signal, but randomises nonlinear 
-properties of the signal (i.e. Hurst exponents)[^Keylock2006]. These surrogates
-can therefore be used to test for changes in nonlinear properties of the 
-original signal.
-
-In contrast to IAAFT surrogates, the IAAFT-wavelet surrogates also 
-preserves nonstationarity. Using other `surromethod`s does not necessarily
-preserve nonstationarity.
-
-To deal with nonstationary signals, Keylock (2006) recommends using a 
-wavelet with a high number of vanishing moments. Thus, the default is to
-use a Daubechies wavelet with 16 vanishing moments.
-
+A wavelet surrogate generated by the following procedure:
+
+1. Compute the wavelet transform of the signal. This results in a set of 
+    detail coefficients over a set of dyadic scales. As in Keylock (2006), 
+    we here use the maximal overlap discrete wavelet transform, or MODWT,
+    so that the number of coefficients at each scale are the same.
+2. Shuffle the detail coefficients at each dyadic scale using the 
+    provided `shufflemethod`. See "Shuffling methods" below for alternatives.
+3. Apply the inverse wavelet transform to the shuffled detail coefficients 
+    to obtain a surrogate time series.
+
+## Shuffling methods
+
+You may choose to use any surrogate from this package to perform the 
+randomization of the detail coefficients at each dyadic scale.
+
+The following methods have been discussed in the literature (more may exist): 
+
+- Random permutations of wavelet coefficients within each scale (Breakspear et al., 2003). To get this behaviour, use `WLS(x, RandomShuffle(), rescale = false, f = nothing)`.
+- Cyclic rotation of wavelet coefficients within each scale (Breakspear et al., 2003).  To get this behaviour, use `WLS(x, Circshift(length(x)), rescale = false, f = nothing)`.
+- Block resampling of wavelet coefficients within each scale (Breakspear et al., 2003). To get this behaviour, use `WLS(x, BlockShuffle(nblocks, randomize = true), rescale = false, f = nothing)`.
+- IAAFT resampling of wavelet coefficients within each scale (Keylock, 2006). To get this behaviour, use `WLS(x, IAAFT(), rescale = true, f = Statistics.cor)`.
+    This method preserves the local mean and variance structure of the signal, but 
```

Comment:
Just check whether this will render correctly in the build docs because it starts with 4 spaces instead of 2. Otherwise, merge!